admin-event-handlers/src/auth/post-confirmation/google_sheets_operations.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::Result;
+use dailp::auth::UserGroup;
+use serde::{Deserialize, Serialize};
+use std::{str::FromStr, time::Duration};
+use tokio::time::sleep;
+
+/// Represents one user's predetermined role.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct UserPermission {
+    /// The user's role
+    pub role: UserGroup,
+    /// The user's group
+    pub email: String,
+}
+
+/// Result obtained directly from the raw Google sheet.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SheetResult {
+    /// Each element here represents one row.
+    pub values: Vec<Vec<String>>,
+}
+
+impl SheetResult {
+    /// Calls the Google Sheets API a number of times.
+    /// Deserializes the first successful result, then stops attempting.
```

Comment:
Is this duplicate code with migrations stuff?

Reply:
See #402